### PR TITLE
Update/review logic

### PR DESCRIPTION
**Pull request by ryckutis**

### CODE REVIEW

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public ApplicationController(ApplicationService applicationService) {
         this.applicationService = applicationService;
     }
 
-    @GetMapping(produces = "application/json")
+    @GetMapping("/all")
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:26]:
```suggestion
    @GetMapping("/applications")
```

File: project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -78,13 +78,47 @@ public ApplicationFormDto create(ApplicationFormDto applicationFormDto) throws U
...
+                throw new UserException("Unsupported user role: " + role);
+        }
+
         application.setStatus(status);
         Application savedApplication = applicationRepository.save(application);
         return applicationMapper.toDto(savedApplication);
     }
 
+    private boolean isValidApplicantStatus(ApplicationStatus currentStatus ,ApplicationStatus newStatus) {
+        return currentStatus == ApplicationStatus.DRAFT && newStatus == ApplicationStatus.PENDING;
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java:111]:
```suggestion
        return Objects.equals(currentStatus,ApplicationStatus.DRAFT) && Objects.equals(newStatus,ApplicationStatus.PENDING);
```

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 3]
```diff
@@ -60,11 +60,11 @@ ResponseEntity<ApplicationFormDto> createApplication(@RequestBody ApplicationFor
     }
 
     @PatchMapping("/{id}")
-    public ResponseEntity<?> updateStatus(@PathVariable("id") long id, @RequestBody ApplicationStatus status) {
+    public ResponseEntity<?> updateStatus(@PathVariable("id") long applicationId, @RequestBody ApplicationStatus status, @RequestHeader("userId") long userId, @RequestHeader("role") UserRole role) {
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:63]: Not the safest way to go about this but we are short on time so let's go with it

File: project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java [hunk 4]
```diff
@@ -78,13 +79,47 @@ public ApplicationFormDto create(ApplicationFormDto applicationFormDto) throws U
...
         Application savedApplication = applicationRepository.save(application);
         return applicationMapper.toDto(savedApplication);
     }
 
+    private boolean isValidApplicantStatus(ApplicationStatus currentStatus ,ApplicationStatus newStatus) {
+        return Objects.equals(currentStatus, ApplicationStatus.DRAFT) && Objects.equals(newStatus, ApplicationStatus.PENDING);
+    }
+
+    private boolean isValidReviewerStatus(ApplicationStatus currentStatus, ApplicationStatus newStatus) {
+        return currentStatus == ApplicationStatus.PENDING && (newStatus == ApplicationStatus.REVIEW_APPROVED || newStatus == ApplicationStatus.REVIEW_DECLINED);
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java:116]: Do `Objects.equals()` and for other cases as well :D

**Review by grkarolis — APPROVED** [review]

File: project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java (hunk 4 above)
Comment (ryckutis) [reply project/src/main/java/com/carlease/project/application/ApplicationServiceImpl.java]: done:D

**Pull request merged**